pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -109,7 +112,7 @@ func (r *AtlasProjectReconciler) teamReconcile(
 				return workflow.OK().ReconcileResult(), nil
 			} else {
 				log.Infow("-> Starting AtlasTeam deletion", "spec", team.Spec)
-				_, err := teamCtx.Client.Teams.RemoveTeamFromOrganization(teamCtx.Context, orgID, team.Status.ID)
+				_, err := teamCtx.SdkClient.TeamsApi.RemoveProjectTeam(teamCtx.Context, projectID, team.Status.ID).Execute()
```

Comment:
I think this changes the logic a bit, no? Teams used to be attached to organizations, not projects. Are we sure here that a team gets deleted from the org, not just detached from a project?

Reply:
Yes you're right! The change in func naming convention that 'remove' no longer refers to deletion misled me. Changing it to delete from org and not just detach from project 👍

Reply:
For this PR it makes sense to keep the functionality, as this is just a refactor for the point of view of end users.

Still, as a general idea, I think we settle that the operator should not be removing Teams, as they might be used for other purposes at the Org level. Should we follow up on this topic?
cc @s-urbaniak

Reply:
@josvazg yes, that would be a good point to come to consensus on prior to the state-machine implementation since it would be a natural point to include that logic -that AKO shouldn't delete a resource on the org level such as teams.